src/Symfony/Component/TypeInfo/Type.php

```diff
@@ -29,61 +24,26 @@ abstract class Type implements \Stringable
 
     public function getBaseType(): BuiltinType|ObjectType
     {
-        if ($this instanceof UnionType || $this instanceof IntersectionType) {
-            throw new LogicException(sprintf('Cannot get base type on "%s" compound type.', (string) $this));
-        }
-
-        $baseType = $this;
-
-        if ($baseType instanceof CollectionType) {
-            $baseType = $baseType->getType();
-        }
-
-        if ($baseType instanceof GenericType) {
-            $baseType = $baseType->getType();
-        }
-
-        return $baseType;
+        return $this;
```

Comment:
I think that this method must be abstract, because we can't make sure that `$this` is a `BuiltinType|ObjectType` in `Type`